src/state/user/users.thunks.ts

```diff
@@ -0,0 +1,46 @@
+import { createAsyncThunk } from '@reduxjs/toolkit'
+
+export type LoginDTO = {
+  username: string
+  password: string
+}
+
+export const mockCredentials: LoginDTO = {
+  username: 'admin',
+  password: 'password',
+}
+
+const login = createAsyncThunk(
+  'user/login',
+  async (credentials: LoginDTO, { rejectWithValue }) => {
+    try {
+      const response = await fetch('http://localhost:9001/api/users/login', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(mockCredentials),
+      })
+
+      if (response.ok) {
+        return await response.json()
+      }
+
+      if (response.status === 401 || response.status === 404) {
+        return rejectWithValue('Invalid credentials')
+      }
+
+      if (response.status >= 500) {
```

Comment:
Can we pool out error response handlers to the separate service? Otherwise you will need to copypast same code in all your GET/POST requests that is realy bad idea.